tasks/community-modules.yml

```diff
@@ -14,17 +14,31 @@
   become: yes
   become_user: "{{ odoo_role_odoo_user }}"
 
+  # Using the output of pip, detect which packages have been installed,
+  # both fresh or upgraded. Save the list with a register so that we can
+  # feed better odoo with it.
 - name: Detect upgraded packages
-  shell: >
-    grep 'Successfully installed' |
-    sed -r 's/ /\n/g' |
-    egrep '^odoo1[0-9]-addon-[^ ]+-[0-9\.]+' |
-    sed -r 's/odoo1[0-9]-addon-([^ ]+)-[0-9\.]+/\1/g' |
-    tr -- '-' '_'
-
   args:
+    # Example input (delete '# ' from each line below)
+    #
+    # Installing collected packages: odoo11-addon-account-financial-report
+    # Found existing installation: odoo11-addon-account-financial-report 11.0.2.4.3
+    # Uninstalling odoo11-addon-account-financial-report-11.0.2.4.3:
+    #  Successfully uninstalled odoo11-addon-account-financial-report-11.0.2.4.3
+    # Successfully installed odoo11-addon-account-financial-report-11.0.2.5.1.99.dev12 odoo11-addon-l10n-es-aeat-mod303-11.0.2.1.0
     stdin: "{{ reg_pip.stdout }}"
     executable: bash
+  # Expected "output" using input from above (delete '# ' from each line below)
+  # note: "output" means `reg_pip_upgraded.stdout`
+  # note: to debug the second regexp you can use https://regexr.com/4l2c3
+  #
+  # account-financial-report
+  # l10n-es-aeat-mod303
+  shell: >
```

Comment:
```suggestion

  # About sed:
  # - option `-r` switches to standard perl regexp, instead of GNU ones. Similar to `egrep` or `grep -E`
  # - option `-n` combined with flag `/p` prints "nothing" except the result of the script. Therefore, unmatching lines are hidden, and matching lines are shown after processed.
  # - capturing groups `(something)` can be referenced inside the matching or at the replacing part with \1 ,  \2, in order or appearence. 
  shell: >
```